npm/pkg/dataplane/dataplane_linux.go

```diff
@@ -21,7 +35,9 @@ func (dp *DataPlane) updatePod(pod *updateNPMPod) error {
 }
 
 func (dp *DataPlane) bootupDataPlane() error {
-	util.DetectIptablesVersion(dp.ioShim)
+	if err := detectIptablesVersion(dp.ioShim); err != nil {
```

Comment:
So at a high level, this tries a bunch of things to figure out the iptables version. I'd suggest making this either be a `[]func(dp.ioShim) error` that you just iterate through, or you can even make some kind of `Detector` interface (I don't think you need that here though... it doesn't seem like each "strategy" holds onto any state that they need.

This gets rid of the awkward `fmt.Sprintf`-ing so you can do it in one place, and you can just return errors from those funcs.

Also, instead of making those funcs take a `dp.io.Shim`... it seems like they all just call `Command`, so you could make them take an interface with just that method... which would make each strategy testable.

Reply:
Thanks for the ideas Tim. I removed that shim argument and converted the function to a method like other places in NPM. In our UTs, we use a mock ioshim to run fake commands.

I got rid of those `fmt.Sprintf` calls too. Realized they weren't needed.

I like the idea of detectors, but since the function isn't _too_ big, I think simplicity wins my favor this time